runtime/runtime/src/bandwidth_scheduler/mod.rs

```diff
@@ -38,55 +48,89 @@ pub fn run_bandwidth_scheduler(
         Some(prev_state) => prev_state,
         None => {
             tracing::debug!(target: "runtime", "Bandwidth scheduler state not found - initializing");
-            BandwidthSchedulerState { mock_data: [0; 32] }
+            BandwidthSchedulerState {
+                link_allowances: Vec::new(),
+                sanity_check_hash: CryptoHash::default(),
+            }
         }
     };
 
-    let scheduler_state_hash: CryptoHash = hash(&borsh::to_vec(&scheduler_state).unwrap());
+    // Prepare the status info for each shard based on congestion info.
+    let mut shards_status: BTreeMap<ShardId, ShardStatus> = BTreeMap::new();
+    for (shard_id, extended_congestion_info) in apply_state.congestion_info.iter() {
+        let last_chunk_missing = extended_congestion_info.missed_chunks_count > 0;
```

Comment:
FYI the missing chunks count is imperfect - it also shows up for missing blocks. I'm not sure if that's relevant here but letting you know just in case.

Reply:
Oh thanks, that's good to know. The only thing that matters for bandwidth scheduler is whether some chunk is missing in the current block. AFAIU `missed_chunks_count > 0` checks for that, even when there are missing blocks.
`block.height() - chunk.height_included()` will always be zero when it's a new chunk in the block, right?